### PR TITLE
refactor(nds-198): refactored dropdown removed base dropdown and base …

### PR DESCRIPTION
**Pull request by Apatil15**
Refactored dropdown with following code chanages. 

1. Added papper js
2. Removed baseListbox and baseOption
3. Added Listbox and option
4. Removed unwanted code.

https://pr-242.d2xq18ppjtz6lz.amplifyapp.com

### CODE REVIEW

File: packages/core/src/components/dropdown/index.scss [hunk 1]
```diff
@@ -49,19 +49,22 @@
...
+	overflow-y: none;
 	list-style-type: none;
 	background-color: var(--nds-background-color);
 	border: var(--nds-dropdown-border-width) solid var(--nds-base-color-40);
-	border-radius: var(--nds-radius-base);
-	box-shadow: var(--nds-dropdown-listbox-shadow);
+}
+
+@mixin listbox {
+	border: unset !important;
```
Comment (sh0ji) [inline packages/core/src/components/dropdown/index.scss:66]: Is there a way to accomplish this without `!important`?

File: packages/react/src/components/Dropdown/index.tsx [hunk 2]
```diff
@@ -1,185 +1,81 @@
...
+	isOpen = false,
 	matchWidth,
 	buttonWidth,
 	sort,
 	baseName = 'nds-dropdown',
 	buttonClass = `${baseName}__button`,
-	listboxClass = `${baseName}__listbox`,
-	optionClass = `${baseName}__option`,
+	popperClass = `${baseName}__popper`,
+	listboxClass = `nds-listbox ${baseName}__listbox`,
```
Comment (sh0ji) [inline packages/react/src/components/Dropdown/index.tsx:34]: If you want to use the `.nds-listbox` styles on the `.nds-dropdown__listbox`, you should use the mixin in the stylesheet to merge the styles and override them. Something like:

```scss
@use '../listbox';

@mixin listbox {
  @include listbox.base;
  @include listbox.group;

  // overrides have to come after the includes
  --nds-listbox-border: unset;
  --nds-listbox-border-radius: unset;
}

...

.nds-dropdown__listbox {
  @include listbox;
}
```

The downside here is that the resulting stylesheet will have duplicated styles, which I don't love. The alternative would be to refactor the listbox styles to separate base styling from more specific use cases (for example, making the border properties default to unset).

File: packages/core/src/components/dropdown/index.scss [hunk 3]
```diff
@@ -49,19 +49,22 @@
...
-@mixin listbox {
+@mixin popper {
 	$-option-computed-height: calc(var(--nds-dropdown-option-font-size) * var(--nds-dropdown-option-line-height) + var(--nds-dropdown-option-padding-y) * 2);
 
 	z-index: var(--nds-zindex-dropdown);
 	max-height: calc(var(--nds-dropdown-options-max-display) * #{$-option-computed-height} + var(--nds-dropdown-border-width) * 2);
 	padding: 0;
 	overflow-x: hidden;
-	overflow-y: auto;
+	overflow-y: none;
```
Comment (sh0ji) [inline packages/core/src/components/dropdown/index.scss:59]: `none` isn't a valid [overflow value](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow#values).

File: packages/react/src/components/Dropdown/types.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+import { PopperOptions } from '../../utilities/popper/types';
+import { FieldInfoCoreProps } from '../Field';
+import { PopperProps } from '../Popper';
+
+type BaseProps = 'children' | 'className' | 'disabled' | 'id';
+
+type PopperInherited = Pick<PopperProps,
+| 'isOpen'
+| 'transition'
+| 'reference'
```
Comment (sh0ji) [inline packages/react/src/components/Dropdown/types.ts:11]: `reference` isn't being used so let's just remove it. If someone needs access to the button, we can talk about adding it then.

File: packages/react/src/components/Dropdown/types.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { PopperOptions } from '../../utilities/popper/types';
+import { FieldInfoCoreProps } from '../Field';
+import { PopperProps } from '../Popper';
+
+type BaseProps = 'children' | 'className' | 'disabled' | 'id';
+
+type PopperInherited = Pick<PopperProps,
+| 'isOpen'
+| 'transition'
```
Comment (sh0ji) [inline packages/react/src/components/Dropdown/types.ts:10]: `transition` should be passed down to the Popper and let's give it a default value of "fade" so it uses the opacity transition by default.

**Review by sh0ji — CHANGES_REQUESTED** [review]
This is what I was hoping to see for this, @Apatil15. Nice job.

I called out a few little things but appreciate the thought you put into it all. After looking closely at everything, I do think the styles could use some refactoring (including listbox styling) but we could probably just do that in a future update.

File: packages/react/src/components/Dropdown/index.tsx [hunk 6]
```diff
@@ -320,37 +259,47 @@ export const Dropdown: DropdownType = ({
...
-				<BaseListbox
+			<Popper
+				placement={placement}
+				className={popperClass}
+				reference={button}
+				isOpen={open}
+				modifiers={modifiers}
+				strategy={strategy}
+				onFirstUpdate={onFirstUpdate}
+				matchWidth={
```
[the remark this answers is not in the document]
Comment (Apatil15) [reply packages/react/src/components/Dropdown/index.tsx:282]: I used transition='fade'  and match with button story was not working as expected I tried all possible scenarios this is one of the them which I forgot to removed.

File: packages/react/src/components/Dropdown/types.ts (hunk 5 above)
Comment (Apatil15) [reply packages/react/src/components/Dropdown/types.ts:10]: I used transition='fade' but removed due to  one of the story MatchButtonWidth not working as expected, option is showing and hiding before match button width, it should be work like production.

File: packages/react/src/components/Dropdown/index.tsx (hunk 2 above)
Comment (Apatil15) [reply packages/react/src/components/Dropdown/index.tsx:34]: Done

File: packages/core/src/components/dropdown/index.scss (hunk 1 above)
Comment (Apatil15) [reply packages/core/src/components/dropdown/index.scss:66]: After included following implement in class no need !important
`@include listbox.base;`
`@include listbox.group;`

File: packages/core/src/components/dropdown/index.scss (hunk 3 above)
Comment (Apatil15) [reply packages/core/src/components/dropdown/index.scss:59]: Done

File: packages/react/src/components/Dropdown/index.tsx [hunk 7]
```diff
@@ -1,185 +1,81 @@
...
 	descriptionId: descIdProp,
 	buttonId: buttonIdProp,
 	listboxId: listboxIdProp,
 	disabled,
-	className,
+	children,
+	placement = 'bottom-start',
+	strategy = 'fixed',
+	onFirstUpdate,
+	modifiers = [
```
[the remark this answers is not in the document]
Comment (Apatil15) [reply packages/react/src/components/Dropdown/index.tsx:55]: Done

**Comment by Apatil15** [comment]
@sh0ji 
Changes has been pushed, please review and confirm 

https://pr-242.d2xq18ppjtz6lz.amplifyapp.com/storybook/?path=/story/dropdown--default

File: packages/react/src/components/Dropdown/types.ts (hunk 5 above)
Comment (Apatil15) [reply packages/react/src/components/Dropdown/types.ts:10]: I have added timeout  to setTransition transition 'fade'.

```js
setTimeout(() => {
setTransition(transitionProp);
}, 500);
```

**Pull request merged**